src/poetry/core/packages/utils/utils.py

```diff
@@ -119,14 +120,22 @@ def strip_extras(path: str) -> tuple[str, str | None]:
     return path_no_extras, extras
 
 
-def is_installable_dir(path: str) -> bool:
-    """Return True if `path` is a directory containing a setup.py file."""
-    if not os.path.isdir(path):
+@functools.lru_cache(maxsize=None)
```

Comment:
I'm curious how much of a hot path this is/if caching is a significant benefit here?

Reply:
My thinking is this is called every time a dir dependency gets cloned, this happens a few times during resolution. And since this requires IO figured it can't hurt.